# HG changeset patch
# User planetmaker <ottd@planetmaker.de>
# Date 1273431190 -7200
# Node ID cf3ff6a68aa819c0faeb9713afdb3839521dd580
# Parent  bcf8c69edafccd8a43a9ca466e68c65c457c5bde
- Codechange: Add the vehicle driving side (also) to the advanced settings

diff -r bcf8c69edafc -r cf3ff6a68aa8 src/lang/english.txt
--- a/src/lang/english.txt	Sun May 09 19:35:17 2010 +0200
+++ b/src/lang/english.txt	Sun May 09 20:53:10 2010 +0200
@@ -1085,6 +1085,9 @@
 STR_CONFIG_SETTING_CATCHMENT                                    :{LTBLUE}Allow more realistically sized catchment areas: {ORANGE}{STRING1}
 STR_CONFIG_SETTING_EXTRADYNAMITE                                :{LTBLUE}Allow removal of more town-owned roads, bridges, etc: {ORANGE}{STRING1}
 STR_CONFIG_SETTING_MAMMOTHTRAINS                                :{LTBLUE}Enable building very long trains: {ORANGE}{STRING1}
+STR_CONFIG_SETTING_DRIVE_SIDE                                   :{LTBLUE}Driving side: {ORANGE}{STRING1}
+STR_CONFIG_SETTING_DRIVE_SIDE_LEFT				:Left
+STR_CONFIG_SETTING_DRIVE_SIDE_RIGHT				:Right
 STR_CONFIG_SETTING_TRAIN_ACCELERATION_MODEL                     :{LTBLUE}Train acceleration model: {ORANGE}{STRING1}
 STR_CONFIG_SETTING_TRAIN_ACCELERATION_MODEL_ORIGINAL            :Original
 STR_CONFIG_SETTING_TRAIN_ACCELERATION_MODEL_REALISTIC           :Realistic
diff -r bcf8c69edafc -r cf3ff6a68aa8 src/settings.cpp
--- a/src/settings.cpp	Sun May 09 19:35:17 2010 +0200
+++ b/src/settings.cpp	Sun May 09 20:53:10 2010 +0200
@@ -979,7 +979,12 @@
 static bool CheckRoadSide(int p1)
 {
 	extern bool RoadVehiclesAreBuilt();
-	return _game_mode == GM_MENU || !RoadVehiclesAreBuilt();
+	if (_game_mode != GM_MENU && RoadVehiclesAreBuilt()) {
+		ShowErrorMessage(STR_CONFIG_SETTING_DYNAMIC_ENGINES_EXISTING_VEHICLES, INVALID_STRING_ID, WL_ERROR);
+		return false;
+	}
+	
+	return true;
 }
 
 /** Conversion callback for _gameopt_settings_game.landscape
diff -r bcf8c69edafc -r cf3ff6a68aa8 src/settings_gui.cpp
--- a/src/settings_gui.cpp	Sun May 09 19:35:17 2010 +0200
+++ b/src/settings_gui.cpp	Sun May 09 20:53:10 2010 +0200
@@ -1357,6 +1357,7 @@
 	SettingEntry(&_settings_vehicles_autorenew_page, STR_CONFIG_SETTING_VEHICLES_AUTORENEW),
 	SettingEntry(&_settings_vehicles_servicing_page, STR_CONFIG_SETTING_VEHICLES_SERVICING),
 	SettingEntry(&_settings_vehicles_trains_page, STR_CONFIG_SETTING_VEHICLES_TRAINS),
+	SettingEntry("vehicle.road_side"),
 	SettingEntry("order.gotodepot"),
 	SettingEntry("gui.new_nonstop"),
 	SettingEntry("gui.order_review_system"),
diff -r bcf8c69edafc -r cf3ff6a68aa8 src/table/settings.h
--- a/src/table/settings.h	Sun May 09 19:35:17 2010 +0200
+++ b/src/table/settings.h	Sun May 09 20:53:10 2010 +0200
@@ -362,7 +362,6 @@
  SDT_CONDOMANY(GameSettings, game_creation.town_name,              SLE_UINT8, 97, SL_MAX_VERSION, 0,NN, 0, 255, _town_names,      STR_NULL,                                  NULL, NULL),
  SDT_CONDOMANY(GameSettings, game_creation.landscape,              SLE_UINT8, 97, SL_MAX_VERSION, 0,NN, 0,   3, _climates,        STR_NULL,                                  NULL, ConvertLandscape),
 	 SDT_CONDVAR(GameSettings, game_creation.snow_line,              SLE_UINT8, 97, SL_MAX_VERSION, 0,NN, DEF_SNOWLINE_HEIGHT * TILE_HEIGHT, MIN_SNOWLINE_HEIGHT * TILE_HEIGHT, DEF_SNOWLINE_HEIGHT * TILE_HEIGHT, 0, STR_NULL,     NULL),
- SDT_CONDOMANY(GameSettings, vehicle.road_side,                    SLE_UINT8, 97, SL_MAX_VERSION, 0,NN, 1,   1, _roadsides,       STR_NULL,                                  CheckRoadSide, NULL),
 
 	    SDT_BOOL(GameSettings, construction.build_on_slopes,                                        0,NN,  true,                    STR_CONFIG_SETTING_BUILDONSLOPES,          NULL),
 	SDT_CONDBOOL(GameSettings, construction.autoslope,                          75, SL_MAX_VERSION, 0, 0,  true,                    STR_CONFIG_SETTING_AUTOSLOPE,              NULL),
@@ -374,6 +373,7 @@
 	SDT_CONDBOOL(GameSettings, economy.allow_town_roads,                       113, SL_MAX_VERSION, 0,NN,  true,                    STR_CONFIG_SETTING_ALLOW_TOWN_ROADS,       NULL),
 	 SDT_CONDVAR(GameSettings, economy.found_town,                   SLE_UINT8,128, SL_MAX_VERSION, 0,MS,TF_FORBIDDEN,TF_BEGIN,TF_END - 1, 1, STR_CONFIG_SETTING_TOWN_FOUNDING, TownFoundingChanged),
 
+	 SDT_CONDVAR(GameSettings, vehicle.road_side,                    SLE_UINT8, 97, SL_MAX_VERSION, 0,MS,     0,     0,       1, 1, STR_CONFIG_SETTING_DRIVE_SIDE, CheckRoadSide),
 	     SDT_VAR(GameSettings, vehicle.train_acceleration_model,     SLE_UINT8,                     0,MS,     0,     0,       1, 1, STR_CONFIG_SETTING_TRAIN_ACCELERATION_MODEL, TrainAccelerationModelChanged),
 	 SDT_CONDVAR(GameSettings, vehicle.roadveh_acceleration_model,   SLE_UINT8,139, SL_MAX_VERSION, 0,MS,     0,     0,       1, 1, STR_CONFIG_SETTING_ROAD_VEHICLE_ACCELERATION_MODEL, RoadVehAccelerationModelChanged),
 	 SDT_CONDVAR(GameSettings, vehicle.train_slope_steepness,        SLE_UINT8,133, SL_MAX_VERSION, 0, 0,     3,     0,      10, 1, STR_CONFIG_SETTING_TRAIN_SLOPE_STEEPNESS,  TrainSlopeSteepnessChanged),
